tchannel-core/src/main/java/com/uber/tchannel/handlers/RequestRouter.java

```diff
@@ -206,6 +216,37 @@ public void channelInactive(ChannelHandlerContext ctx) throws Exception {
         }
     }
 
+    private ListenableFuture<Response> sendRequestToAsyncHandler(
+            final AsyncRequestHandler asyncHandler, final Request request) {
+        // span used to trace this request
+        final Span span = Tracing.startInboundSpan(request,
+            topChannel.getTracer(), topChannel.getTracingContext());
```

Comment:
The `Tracer` and `TracingContext` are only present when the channel is created with them, so you'll have to check whether the `Tracer` is null and fallback to the previous behavior.